### PR TITLE
HRT-438 - changing status

### PR DESCRIPTION
**Pull request by gdbarnes**

### CODE REVIEW

File: components/admin/actions.tsx [hunk 1]
```diff
@@ -267,6 +51,15 @@ export default function Actions({ data }: PageProps): JSX.Element {
         'aboveZero',
         'Bedroom need should be a number greater than 0',
         (value, testContext) => {
+          console.log(value, testContext);
```
Comment (angry-dan) [inline components/admin/actions.tsx:54]: debug!

File: components/admin/actions.tsx [hunk 2]
```diff
@@ -267,6 +51,15 @@ export default function Actions({ data }: PageProps): JSX.Element {
...
+
+          if (
+            testContext.parent.status !== ApplicationStatus.ACTIVE &&
+            testContext.parent.status !== ApplicationStatus.ACTIVE_UNDER_APPEAL
+          ) {
+            return true;
```
Comment (angry-dan) [inline components/admin/actions.tsx:60]: This validation condition doesn't seem to match the error message: "Bedroom need should be a number greater than 0"

File: components/admin/actions.tsx [hunk 3]
```diff
@@ -341,94 +134,138 @@ export default function Actions({ data }: PageProps): JSX.Element {
...
     }
 
-    updateApplication(request);
-    setTimeout(() => router.reload(), 500);
+    try {
+      updateApplication(request).then(() => {
+        router.reload();
+      });
+    } catch (error) {
+      console.log('error', error);
```
Comment (angry-dan) [inline components/admin/actions.tsx:142]: what's the benefit of catching this error? For context I'm setting up Sentry monitoring at the moment which will notice when errors get thrown and record them centrally.

File: pages/applications/view/[id]/index.tsx [hunk 4]
```diff
@@ -56,8 +56,28 @@ export default function ApplicationPage({
   data,
   history,
 }: PageProps): JSX.Element | null {
-  if (!data.id) return <Custom404 />;
   const router = useRouter();
```
Comment (angry-dan) [inline pages/applications/view/[id]/index.tsx:59]: I'm pretty sure that `useRouter` is a state based hook, which means that every time that `router.query.tab` changes, this component will be re-rending anyway. If I'm right, then I think you could skip the state all together and just do:

```
  const tab = router.query.tab ?? 'overview';

  const handleTabChange = (newValue: string) => {
    router.push(`/applications/view/${data.id}?tab=${newValue}`, undefined, {
      shallow: true,
    });
  };
```

And each time you call handleTabChange you should be able to rely on tab updating. In other words, router.push is internally doing a setState so you probably don't need a second one yourself.

**Review by angry-dan — APPROVED** [review]
Happy to approve but a couple of observations. Up to you if you need to make changes.

File: components/admin/actions.tsx (hunk 2 above)
Comment (gdbarnes) [reply components/admin/actions.tsx:60]: The idea here is to skip validation (return true) when the selected status doesn't show the bedroom need question. Maybe there's a better way but I'm not sure what it is.

File: components/admin/actions.tsx (hunk 3 above)
Comment (gdbarnes) [reply components/admin/actions.tsx:142]: Will remove

File: components/admin/actions.tsx (hunk 1 above)
Comment (gdbarnes) [reply components/admin/actions.tsx:54]: Will remove

File: pages/applications/view/[id]/index.tsx (hunk 4 above)
Comment (gdbarnes) [reply pages/applications/view/[id]/index.tsx:59]: Makes sense. Have updated. We could probably use this shallow routing on other places in the admin but maybe for another PR.

**Pull request merged**